### ISSUE TITLE
`consistent-function-scoping` fails, even though function uses block-scoped value

### ISSUE THREAD

**Issue by RyanZim** [sindresorhus/eslint-plugin-unicorn#2164]
`unicorn/consistent-function-scoping` demands that functions that be moved to the outer scope, even though the function captures a block-scoped variable.

Totally contrived example:
```js
function foo() {
  for (let i = 0; i < 3; i++) {
    const scoped = i + 42
    const fn = () => 5 + scoped
    console.log(fn())
  }
}
```
results in:
```
4:19  error  Move arrow function 'fn' to the outer scope  unicorn/consistent-function-scoping
```
Of course, it's impossible to move `fn` to the outer scope, as `scoped` isn't defined there.

**Comment by Samuel-Therrien-Beslogic** [comment]
Could this be related to https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2088, where in that issue the outside scope value is `this` ?

**Comment by RyanZim** [comment]
Indeed; this does seem to be a duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/792; missed that one when searching apparently.
